### PR TITLE
check if self.source is an Image; add test_fastsam_text_prompt

### PR DESCRIPTION
**Pull request by abrichr**
Addresses https://github.com/ultralytics/ultralytics/issues/13218:

- checks if `self.source` is a `PIL.Image` before `os.path.isdir`
- adds a test

### CODE REVIEW

**Comment by abrichr** [comment]
I have read the CLA Document and I sign the CLA

**Comment by abrichr** [comment]
@glenn-jocher can you please suggest how to fix the failing check?

```
.github/ISSUE_TEMPLATE/question.yml 3ms (unchanged)
.github/ISSUE_TEMPLATE/config.yml 2ms (unchanged)
.github/ISSUE_TEMPLATE/bug-report.yml 10ms (unchanged)
.github/dependabot.yml 3ms (unchanged)
Run codespell \
  codespell \
    --write-changes \
    --ignore-words-list "crate,nd,ned,strack,dota,ane,segway,fo,gool,winn,commend,bloc,nam,afterall" \
    --skip "*.pt,*.pth,*.torchscript,*.onnx,*.tflite,*.pb,*.bin,*.param,*.mlmodel,*.engine,*.npy,*.data*,*.csv,*pnnx*,*venv*,*translat*,__pycache__*,*.ico,*.jpg,*.png,*.mp4,*.mov,/runs,/.git,./docs/??/*.md,./docs/mkdocs_??.yml"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
Run git add .
  git add .
  git reset HEAD -- .github/workflows/  # workflow changes are not permitted with default token
  if ! git diff --staged --quiet; then
    git config --global user.name "UltralyticsAssistant"
    git config --global user.email "web@ultralytics.com"
    git commit -m "Auto-format by https://ultralytics.com/actions"
    git push
  else
    echo "No changes to commit"
  fi
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
[fix/fastsam-text-prompt 610870b9] Auto-format by https://ultralytics.com/actions
 1 file changed, 1 insertion(+), 3 deletions(-)
remote: Permission to OpenAdaptAI/ultralytics.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/OpenAdaptAI/ultralytics/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```

**Comment by glenn-jocher** [comment]
@abrichr thanks for the PR! Test function won't work like this though, it needs defined arguments. You can use the BUS constant or ASSETS / bus.jpg for example to load a PIL image of bus.jpg for the test.

File: tests/test_python.py [hunk 1]
```diff
@@ -576,3 +576,18 @@ def test_yolo_world():
         close_mosaic=1,
         trainer=WorldTrainerFromScratch,
     )
+
+
+def test_fastsam_text_prompt(image, model_name="FastSAM-s.pt", device="cpu"):
+    """Test FastSAM with FastSAMPrompt.text_prompt."""
+    model = FastSAM(model_name)
```
Comment (Burhan-Q) [inline tests/test_python.py:583]: Looks like you might have forgotten to add an import for `FastSAM` at the top of the file, which is why all the tests are currently failing.

```py
    def test_fastsam_text_prompt(image, model_name="FastSAM-s.pt", device="cpu"):
        """Test FastSAM with FastSAMPrompt.text_prompt."""
>       model = FastSAM(model_name)

E      NameError: name 'FastSAM' is not defined
```

**Comment by glenn-jocher** [comment]
@abrichr yes I would just delete your test and update the existing test in tests/test_cli.py

```python
import pytest
from pathlib import Path
from PIL import Image
from ultralytics import run, checks

@pytest.mark.skipif(checks.IS_PYTHON_3_12, reason="MobileSAM with CLIP is not supported in Python 3.12")
def test_fastsam(task="segment", model=Path("WEIGHTS_DIR/FastSAM-s.pt"), data="coco8-seg.yaml"):
    """Test FastSAM segmentation functionality within Ultralytics."""
    source = Path("ASSETS/bus.jpg")

    run(f"yolo segment val {task} model={model} data={data} imgsz=32")
    run(f"yolo segment predict model={model} source={source} imgsz=32 save save_crop save_txt")

    from ultralytics import FastSAM
    from ultralytics.models.fastsam import FastSAMPrompt
    from ultralytics.models.sam import Predictor

    # Create a FastSAM model
    sam_model = FastSAM(model)  # or FastSAM-x.pt

    # Run inference on an image file path
    everything_results = sam_model(source, device="cpu", retina_masks=True, imgsz=1024, conf=0.4, iou=0.9)

    # Check inference with PIL image
    _ = sam_model(Image.open(source), device="cpu", retina_masks=False, imgsz=1024, conf=0.4, iou=0.9)

    # Remove small regions
    new_masks, _ = Predictor.remove_small_regions(everything_results[0].masks.data, min_area=20)

    # Everything prompt
    prompt_process = FastSAMPrompt(source, everything_results, device="cpu")
    ann = prompt_process.everything_prompt()

    prompt_process_pil = FastSAMPrompt(pil_image, everything_results_pil, device="cpu")

    # Bbox default shape [0,0,0,0] -> [x1,y1,x2,y2]
    ann = prompt_process.box_prompt(bbox=[200, 200, 300, 300])

    # Text prompt
    ann = prompt_process.text_prompt(text="a photo of a dog")

    # Point prompt
    # Points default [[0,0]] [[x1,y1],[x2,y2]]
    # Point_label default [0] [1,0] 0:background, 1:foreground
    ann = prompt_process.point_prompt(points=[[200, 200]], pointlabel=[1])

    prompt_process.plot(annotations=ann, output="./")
```

**Comment by glenn-jocher** [comment]
Normally I'd do this myself but there's some permissions issues going on with your fork, they seem to be affecting the formatting actions too.

**Comment by abrichr** [comment]
> Test function won't work like this though, it needs defined arguments

I was re-using the `image` fixture, but I took your suggestion and replaced the test with your changes.

**Comment by glenn-jocher** [comment]
@abrichr merged into docs branch for further work. Thank you for your contributions!

**Pull request merged**